### PR TITLE
refactor: remove unnecessary enabled addon config state representation

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

The definitive configuration state representation for addons is in the `KubernetesConfig` `Addons` array. This PR removes the unnecessary setting of an `isEnabled` property during addons data marshalling, and instead consumes the authoritative `Enabled` boolean in the `Addons` config array.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/artifacts.go [hunk 1]
```diff
@@ -36,133 +36,111 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
 			sourceFile:      "kubernetesmasteraddons-heapster-deployment.yaml",
 			base64Data:      k.GetAddonScript(HeapsterAddonName),
 			destinationFile: "kube-heapster-deployment.yaml",
-			isEnabled:       k.IsAddonEnabled(HeapsterAddonName),
```
Comment (jackfrancis) [inline pkg/engine/artifacts.go]: Upside: we remove duplicate code!

File: pkg/engine/artifacts_test.go [hunk 2]
```diff
@@ -12,339 +12,6 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
-func TestKubernetesContainerAddonSettingsInit(t *testing.T) {
```
Comment (jackfrancis) [inline pkg/engine/artifacts_test.go]: This UT essentially tests the functionality that we're removing, so it goes.

File: pkg/engine/artifacts.go (hunk 1 above)
Comment (CecileRobertMichon) [reply pkg/engine/artifacts.go]: do we still need `isEnabled` to be part of `kubernetesComponentFileSpec` or is it removed everywhere?

Comment (CecileRobertMichon) [reply pkg/engine/artifacts.go]: nevermind it's still used in `kubernetesAddonSettingsInit`. What's the difference with `kubernetesContainerAddonSettingsInit` and why remove it in one but not both?

Comment (jackfrancis) [reply pkg/engine/artifacts.go]: It's still being used by the non-user-configurable addons, and the manifests.

I'm scaffolding out a refactor plan to move all addons into the "user-configurable addons" interface, and to do something similar for the `/etc/k8s/manifests` files as well, after which time we can possibly remove the property altogether.

Comment (jackfrancis) [reply pkg/engine/artifacts.go]: The difference is that `kubernetesContainerAddonSettingsInit` is able to correlate with an `KubernetesAddon` item in the `Addons` array.

`kubernetesAddonSettingsInit`, on the other hand, *is* where the authoritative state of whether or not a particular "non-user-configurable addon" is enabled or not. (This is tech debt.)

**Comment by jackfrancis** [comment]
Related to #2251

**Pull request merged**